library/Fission/CLI/Auth.hs

```diff
@@ -35,10 +45,14 @@ cachePath = do
   home <- getHomeDirectory
   return $ home </> ".fission.yaml"
 
-withAuth :: (MonadRIO cfg m, HasLogFunc cfg) => (BasicAuthData -> m ()) -> m ()
+withAuth :: (MonadRIO cfg m, HasLogFunc cfg)
+         => (BasicAuthData -> m (Either ClientError a))
+         -> m (Either CLIError a)
 withAuth action = get >>= \case
   Right auth ->
-    action auth
+    action auth >>= \case
+      Right result -> return $ Right result
+      Left err -> return . Left . CLIError . UTF8.textShow $ err
```

Comment:
If you want to, those `return`s can be moved before the `case`. This is specifically because it's happening on both branches:

```haskell
action auth >>= pure . \case
  Right result -> Right result
  Left err -> Left . CLIError . UTF8.textShow $ err
```

Which, per the `Bifunctor` comment above could also be:

```haskell
action auth >>= pure . first (CLIError . UTF8.textShow)
```

...but we're not playing code golf 🏌️‍♀️

Reply:
HLint doesn't like that first option (still compiles fine), and recommends:
```haskell
    \case
      Right result -> Right result
      Left err -> Left . CLIError . UTF8.textShow $ err
      <$> action auth
```
which I'm not crazy about because the flow isn't super clear